plugin-server/src/worker/ingestion/event-pipeline/runner.ts

```diff
@@ -80,116 +50,164 @@ export class EventPipelineRunner {
     // hitting Postgres in the capture endpoint. Eventually the entire pipeline should
     // follow this route and we can rename it to just be `runEventPipeline`.
     async runLightweightCaptureEndpointEventPipeline(event: PipelineEvent): Promise<EventPipelineResult> {
-        this.hub.statsd?.increment('kafka_queue.lightweight_capture_endpoint_event_pipeline.start', {
+        this.hub.statsd?.increment('kafka_queue.event_pipeline.start', {
             pipeline: 'lightweight_capture',
         })
-        const result = await this.runPipeline('populateTeamDataStep', event)
-        this.hub.statsd?.increment('kafka_queue.single_event.processed_and_ingested')
-        return result
-    }
 
-    async runEventPipeline(event: PluginEvent): Promise<EventPipelineResult> {
-        this.hub.statsd?.increment('kafka_queue.event_pipeline.start', { pipeline: 'event' })
-        const result = await this.runPipeline('emitToBufferStep', event)
-        this.hub.statsd?.increment('kafka_queue.single_event.processed_and_ingested')
-        return result
+        try {
+            const eventWithTeam = await this.runStep(populateTeamDataStep, [this, event])
+            if (eventWithTeam != null) {
+                const result = await this.runEventPipelineSteps(eventWithTeam)
+                this.hub.statsd?.increment('kafka_queue.single_event.processed_and_ingested')
+                return result
+            } else {
+                return this.registerLastStep('populateTeamDataStep', null, [event])
```

Comment:
we should be doing `this.hub.statsd?.increment('kafka_queue.single_event.processed_and_ingested')` always, no?